addons/pyroscope/readme.md

```diff
@@ -0,0 +1,195 @@
+# Pyroscope
+
+This addon is built based [Pyroscope](https://github.com/pyroscope-io/pyroscope), which consist of server and agent. It allows the user to collect, store, and query the profiling data in a CPU and disk efficient way.
+
+## install
+
+```shell
+vela addon enable pyroscope
+```
+
+## uninstall
+
+```shell
+vela addon disable pyroscope
+```
+
+## Note
+### Pyroscope for Golang applications
+
+- To start profiling a Go application, you need to include our go module in your app
+    ```shell
+    # make sure you also upgrade pyroscope server to version 0.3.1 or higher
+    go get github.com/pyroscope-io/client/pyroscope
+    ```
+- Then add the following code to your application:
+```go
+package main
+
+import "github.com/pyroscope-io/client/pyroscope"
+
+func main() {
+  pyroscope.Start(pyroscope.Config{
+    ApplicationName: "simple.golang.app",
+    // replace this with the address of pyroscope server
+    ServerAddress:   "http://pyroscope-server:4040",
+    // you can disable logging by setting this to nil
+    Logger:          pyroscope.StandardLogger,
+
+    // optionally, if authentication is enabled, specify the API key:
+    // AuthToken: os.Getenv("PYROSCOPE_AUTH_TOKEN"),
+
```

Comment:
I think developer can read these configuration from env. Then we can support a trait patch these env to a container.
Then app use this trait by
```yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: first-vela-app
spec:
  components:
    - name: express-server
      type: webservice
      properties:
        image: oamdev/hello-world
        ports:
         - port: 8000
           expose: true
      traits:
        - type: scaler
          properties:
            replicas: 1
        - type: pyroscope
           server: "http://pyroscope-server:4040"
           appName: context.name
           logger: "pyroscope.StandardLogger"
           ....
```

Reply:
Yes, it's a convenient way for user. I modify it. @wangyikewxgm